src/common/portal-metadata-context/PortalMetadataContext.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { IContextItem, IPortalMetadata } from "./Interfaces";
+
+export class PortalMetadataContext {
+
+    private portalMetadata: IPortalMetadata;
+    private portalMetadataContext: IContextItem;
+
+
+    public getPortalMetadataContext() {
+        // returns portalMetadataContext
+    }
+
+    public onFileUpdate() {
+        // updates metadatacontext when file updates
+    }
+
+    private updatePortalMetadataContext() {
+        // updates portalMetadataContext
+    }
+
+    private updatePortalMetadata() {
+        // updates portalMetadata
+    }
+
+    private createPortalMetadataContext() {
```

Comment:
create is not the right terminology - lets use get/set/update for both of these members and use singleton class as suggested above.